bloom-housing/bloom
# 1381/listings management, application addresses

# Pull Request Template

## Issue

Addresses #1381

## Description

Adds the application addresses section to the listings management prototype. This section is probably 10x the work and size of any other section so far so it might be harder to review 😓.

The new backend fields we needed are: `applicationMailingAddress` `applicationDropOffAddress` `applicationDropOffAddressOfficeHours` `applicationPickUpAddressType` `applicationDropOffAddressType` `additionalApplicationSubmissionNotes`

Here are the [Figma](https://www.figma.com/file/Y8bFd0aSz7eYJ2TCNid6s3/bloom-partners?node-id=2726%3A126111) states. Visually my sections are not arranged in exactly the same way because our form grids are not responsive to mobile in the exact layout in the designs. You have a leasing agent address section, a mailing address checkbox that toggles on the mailing address field, a radio button for application pickup that toggles an application pickup location radio group wherein if you select another address it toggles on a new address field, and the same as for application drop off. Lastly you have a postmark radio that if selected toggles on a postmark date field. Application submission notes is new and not currently used anywhere in the public app.

The issue included removing the PoBox application method and I went ahead and removed the LeasingAgent and PaperPickup methods as well since they were used identically to the PoBox method and seemed really unnecessary.

## Type of change


## How Can This Be Tested/Reviewed?

Oh boy. I would create a few listings from listings/add in partners doing a number of different combinations of possible inputs and then checking them still within partners on the details page to ensure the data is showing appropriately w all the toggleable options. Mailing address should only show as a radio option if you have indicated that a mailing address should exist with the checkbox.

The biggest thing I would test is then re-editing a listing you just made and making sure all the default values load in appropriately with what you just submitted (I had a number of bugs here). Include toggling sections on and off again after having the form loaded with default values to make sure all is working appropriately.

I would also test adding all the addresses and then removing them and making sure they are actually removed (had bugs here).

From the public side, I would create a custom leasing agent address, mailing address, pickup address, and dropoff address and make sure they all appear. I would create one with no addresses and make sure that looks alright. And lastly I would select the dropoff and pickup to connect to the leasing and mailing addresses to make sure those pull through.

## Checklist:
